src/ripple/protocol/Protocol.h

```diff
@@ -67,7 +67,7 @@ std::size_t constexpr maxDeletableDirEntries = 1000;
 std::size_t constexpr maxTokenOfferCancelCount = 500;
 
 /** The maximum number of offers in an offer directory for NFT to be burnable */
-std::size_t constexpr maxDeletableTokenOfferEntries = 500;
+int constexpr maxDeletableTokenOfferEntries = 500;
```

Comment:
Nitpick. Preferrably, we want an unsigned data type here so that the compiler automatically spots a mistake like
`size_t constexpr maxDeletableTokenOfferEntries = -500;`

Reply:
https://github.com/XRPLF/rippled/pull/4346#discussion_r1072849839
It was discussed here that to allow negative numbers. Because otherwise, the line 
` if (maxDeletableTokenOfferEntries - deletedBuyOffers > 0)`
does not really make sense

Reply:
For unsigned type, we can write
```if (maxDeletableTokenOfferEntries > deletedBuyOffers)```

It seems strange to use a signed type to store a non-negative value.

Reply:
cool will change

Reply:
FWIW, the change to `int` was prompted by a bug in the original code that involved:

1. unsigned numbers,
2. a subtraction and
3. a comparison for `<=`

The original code ended up, unintentionally, being the equivalent of a `while (true)`.  The bug was a prime illustration of the hazards of unsigned integers in the hands of the unwary.   This hazard is further aggravated by the presence of `auto`.  A very straight forward fix was to convert from using unsigned to signed arithmetic. 

There are sporadic debates within the standards committee where certain influential voices shout that we should be using signed arithmetic as many places as possible.  They champion signed arithmetic precisely because of bugs like the one described above.  Other voices argue that we should consistently use unsigned where unsigned is already being used.

I'm not gonna stick my nose any further into this discussion, other than to provide that background.  I'm sure that @drlongle's changes are a fine way to approach the problem.